### PR TITLE
 edit / show title prop: add 'false' and callback choices

### PR DESCRIPTION
**Pull request by hiaselhans**
follow-up from #9768 to change branch

This PR adds two more choices to the 'title' prop of edit / show views:

- false to display no title
- callback with signature (arg0?: RaRecord) => string

I found this to be the most flexible and efficient way to inject a title

### CODE REVIEW

File: packages/ra-ui-materialui/src/detail/Show.spec.tsx
Comment (slax57) [inline packages/ra-ui-materialui/src/detail/Show.spec.tsx:139]: **question:** ‏Why not add tests for Edit too?

File: packages/ra-ui-materialui/src/detail/ShowView.tsx
Comment (slax57) [inline packages/ra-ui-materialui/src/detail/ShowView.tsx:70]: **question:** ‏Why did you need that change?

File: packages/ra-ui-materialui/src/types.ts
Comment (slax57) [inline packages/ra-ui-materialui/src/types.ts:39]: **polish:** ‏

```suggestion
        | ((record: RecordType | undefined) => string);
```

File: packages/ra-ui-materialui/src/types.ts
Comment (slax57) [inline packages/ra-ui-materialui/src/types.ts:84]: **polish:** ‏

```suggestion
        | ((record: RecordType | undefined) => string);
```

**Review by slax57 — CHANGES_REQUESTED** [review]
Thank you for your contribution :blush: 

Two more things I'd like to add to the following comments:
- I saw the following warning in the console while testing: `Warning: Failed prop type: Invalid prop `title` supplied to `Show`, expected a ReactNode.`. You may have forgotten to update a propTypes declaration somewhere.
- I'd like to have the approval of @fzaninotto or @djhi on the feature before merging. Guys wdyt?

**Comment by slax57** [comment]
Got the approval! :tada: 

Also, realizing this will also fix https://github.com/marmelab/react-admin/issues/9695

**Comment by fzaninotto** [comment]
Any updates on this one?

**Comment by fzaninotto** [comment]
Bump, you need to rebase and take the review into account.

**Comment by fzaninotto** [comment]
No news for some time, closing.

**Pull request closed without merging**
